instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/EvOfflineTest.kt

```diff
@@ -0,0 +1,153 @@
+import com.mapbox.navigation.instrumentation_tests.utils.tiles.OfflineRegions
+import com.mapbox.navigation.instrumentation_tests.utils.tiles.withMapboxNavigationAndOfflineTilesForRegion
+import com.mapbox.navigation.instrumentation_tests.utils.withMapboxNavigation
+import com.mapbox.navigation.instrumentation_tests.utils.withoutInternet
+import com.mapbox.navigation.testing.ui.BaseCoreNoCleanUpTest
+import com.mapbox.navigation.testing.ui.utils.coroutines.NavigationRouteAlternativesResult
+import com.mapbox.navigation.testing.ui.utils.coroutines.RouteRequestResult
+import com.mapbox.navigation.testing.ui.utils.coroutines.alternativesUpdates
+import com.mapbox.navigation.testing.ui.utils.coroutines.getSuccessfulResultOrThrowException
+import com.mapbox.navigation.testing.ui.utils.coroutines.requestRoutes
+import com.mapbox.navigation.testing.ui.utils.coroutines.routesUpdates
+import com.mapbox.navigation.testing.ui.utils.coroutines.sdkTest
+import com.mapbox.navigation.testing.ui.utils.coroutines.setNavigationRoutesAsync
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.flow.filterIsInstance
+import kotlinx.coroutines.flow.first
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotEquals
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+
+class EvOfflineTest : BaseCoreNoCleanUpTest() {
+
+    override fun setupMockLocation(): Location {
+        return mockLocationUpdatesRule.generateLocationUpdate {
+            longitude = 13.361378213031003
+            latitude = 52.49813341962201
+        }
+    }
+
+    @Before
+    fun setupMockRoutes() {
+        val evRouteRequestHandler = MockDirectionsRequestHandler(
+            profile = DirectionsCriteria.PROFILE_DRIVING_TRAFFIC,
+            jsonResponse = readRawFileText(context, R.raw.ev_routes_berlin),
+            expectedCoordinates = evRouteInBerlin().coordinatesList(),
+        )
+        mockWebServerRule.requestHandlers.add(evRouteRequestHandler)
+    }
+
+    @Test
+    fun requestRouteWithoutInternetAndTiles() = sdkTest {
+        withMapboxNavigation { navigation ->
+            withoutInternet {
+                val routes = navigation.requestRoutes(evRouteInBerlin())
+                assertTrue(routes is RouteRequestResult.Failure)
+            }
+        }
+    }
+
+    @Test
+    fun startTripWithoutInternetThenTurnItOn() = sdkTest {
+        withMapboxNavigationAndOfflineTilesForRegion(
+            OfflineRegions.Berlin
+        ) { navigation ->
+            navigation.startTripSession()
+            withoutInternet {
+                val requestResult = navigation.requestRoutes(evRouteInBerlin())
+                    .getSuccessfulResultOrThrowException()
+                assertEquals(RouterOrigin.Onboard, requestResult.routerOrigin)
+                navigation.setNavigationRoutesAsync(requestResult.routes)
+            }
+            stayOnPosition(
+                longitude = 13.361378213031003,
```

Comment:
Can you add a comment specifying what this location means? Or just use something that says it (like `routeOptions.coordinatesList.first()`).

Reply:
added